app/src/app.js

```diff
@@ -0,0 +1,157 @@
+/*
+ * Welcome to the new js2coffee 2.0, now
+ * rewritten to use the esprima parser.
+ * try it out!
+ */
+var app, bodyParser, clear, config, connman, device, error, express, hotspot, manager, networkManager, retry, run, ssids, systemd, wifiScan;
+
+express = require('express');
+
+bodyParser = require('body-parser');
+
+connman = require('./connman');
+
+hotspot = require('./hotspot');
+
+networkManager = require('./networkManager');
+
+systemd = require('./systemd');
+
+wifiScan = require('./wifi-scan');
+
+config = require('./config');
+
+app = express();
+
+app.use(bodyParser.json());
+
+app.use(bodyParser.urlencoded({
+  extended: true
+}));
+
+app.use(express["static"](__dirname + '/public'));
+
+ssids = [];
+
+error = function(e) {
+  console.log(e);
+  if (retry) {
+    console.log('Retrying');
+    console.log('Clearing credentials');
+    return manager.clearCredentials().then(run)["catch"](error);
+  } else {
+    console.log('Not retrying');
+    console.log('Exiting');
+    return process.exit();
+  }
+};
+
+app.get('/ssids', function(req, res) {
+  return res.json(ssids);
+});
+
+app.post('/connect', function(req, res) {
+  if (!((req.body.ssid != null) && (req.body.passphrase != null))) {
+    return res.sendStatus(400);
+  }
+  res.send('OK');
+  return hotspot.stop(manager, device).then(function() {
+    return manager.setCredentials(req.body.ssid, req.body.passphrase);
+  }).then(run)["catch"](error);
+});
+
+app.use(function(req, res) {
+  return res.redirect('/');
+});
+
+run = function() {
+  return manager.isSetup().then(function(setup) {
+    if (setup) {
+      console.log('Credentials found');
```

Comment:
Can we be more explicit with the user credentials here? We should print to screen.